lib/pick.js

```diff
@@ -2,7 +2,30 @@
 
 module.exports = function(obj, keys) {
   return keys.reduce((accum, key) => {
-    accum[key] = obj[key];
+    if (!key.includes('.')) {
+      // Regular keys, just add them
+      accum[key] = obj[key];
+      return accum;
+    }
+
+    // Complex keys, let's split them out
+    const [parent, child] = key.split('.');
+
+    // No need to add if it's already adding the whole parent property.
+    if (keys.includes(parent)) {
+      return accum;
+    }
+
+    // Fail-safe bail from undefined property
+    if (!obj[parent]) {
```

Comment:
Are there any `test.*` properties that may have boolean values? (Not including deprecated)

Reply:
Not that I know. Anyway, I only want this to work when parent is an object, so it should always skip any negative value.

I could this to only pass when typeof returns object, if you prefer.

Reply:
Oh right, got it. Thanks